zabbix/helper.go

```diff
@@ -0,0 +1,88 @@
+package provider
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/claranet/go-zabbix-api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func randStringNumber(length int) string {
+	const charset = "0123456789"
+	str := make([]byte, length)
+
+	for i := 0; i < length; i++ {
+		str[i] = charset[rand.Intn(len(charset))]
+	}
+	return string(str)
+}
+
+func sqlError(err error) bool {
+	if strings.Contains(err.Error(), "SQL statement execution") || strings.Contains(err.Error(), "DBEXECUTE_ERROR") {
+		return true
+	}
+	return false
+}
+
+type deleteFunc func([]string) ([]interface{}, error)
```

Comment:
I think `deleteFunc` should be modeled like `createFunc` and accept a pointer to an instance of `zabbix.API` as a parameter.

Reply:
Let's leave this as-is for now.